ovn-tester/tests/network_policy_cross.py

```diff
@@ -0,0 +1,38 @@
+from collections import namedtuple
+from ovn_context import Context
+from ovn_workload import Namespace
+
+NpCrossNsCfg = namedtuple('NpCrossNsCfg',
+                          ['n_ns',
+                           'pods_ns_ratio'])
+
+class NetworkPolicyCross(object):
+    def __init__(self, config):
+        self.config = NpCrossNsCfg(
+            n_ns=config.get('n_ns', 0),
+            pods_ns_ratio=config.get('pods_ns_ratio', 0),
+        )
+
+    def run(self, ovn, global_cfg):
+        all_ns = []
+
+        with Context('network_policy_cross_startup', brief_report=True) as ctx:
+            ports = ovn.provision_ports(
+                    self.config.pods_ns_ratio*self.config.n_ns)
+            for i in range(self.config.n_ns):
+                ns = Namespace(ovn, f'NS_{i}')
+                ns.add_ports(ports[i*self.config.pods_ns_ratio :
+                                   (i + 1) *self.config.pods_ns_ratio])
+                all_ns.append(ns)
+
+        with Context('network_policy', self.config.n_ns) as ctx:
+            for i in ctx:
+                ns = all_ns[i]
+                ext_ns = all_ns[(i+1) % self.config.n_ns]
+                ns.allow_cross_namespace(ext_ns)
```

Comment:
We don't check if the policy is actually enforced.

We probably need something like:
```
diff --git a/ovn-tester/ovn_workload.py b/ovn-tester/ovn_workload.py
index 136952e..5f66688 100644
--- a/ovn-tester/ovn_workload.py
+++ b/ovn-tester/ovn_workload.py
@@ -466,6 +466,14 @@ class Namespace(object):
             worker = dst.metadata
             worker.ping_external(self.cluster, dst)
 
+    @ovn_stats.timeit
+    def check_enforcing_cross_ns(self, ns):
+        if len(self.ports) > 0 and len(ns.ports) > 0:
+            dst = ns.ports[0]
+            src = self.ports[0]
+            worker = src.metadata
+            worker.ping_port(self.cluster, src, dst.ip)
+
 
 class Cluster(object):
     def __init__(self, central_node, worker_nodes, cluster_cfg, brex_cfg):
diff --git a/ovn-tester/tests/network_policy_cross.py b/ovn-tester/tests/network_policy_cross.py
index 4128ff8..87993de 100644
--- a/ovn-tester/tests/network_policy_cross.py
+++ b/ovn-tester/tests/network_policy_cross.py
@@ -23,6 +23,7 @@ class NetworkPolicyCross(object):
                 ns = Namespace(ovn, f'NS_{i}')
                 ns.add_ports(ports[i*self.config.pods_ns_ratio :
                                    (i + 1) *self.config.pods_ns_ratio])
+                ns.default_deny()
                 all_ns.append(ns)
 
         with Context('network_policy', self.config.n_ns) as ctx:
@@ -30,6 +31,7 @@ class NetworkPolicyCross(object):
                 ns = all_ns[i]
                 ext_ns = all_ns[(i+1) % self.config.n_ns]
                 ns.allow_cross_namespace(ext_ns)
+                ns.check_enforcing_cross_ns(ext_ns)
 
         if not global_cfg.cleanup:
             return
```